src/main/java/com/ibm/stocator/fs/swift/auth/JossAccount.java

```diff
@@ -109,4 +119,27 @@ public Account getAccount() {
     }
     return mAccount;
   }
+
+  /**
+   * Creates a thread safe HttpClient to use for requests
+   */
+  private HttpClient initHttpClient() {
+    PoolingHttpClientConnectionManager manager = new PoolingHttpClientConnectionManager();
+    manager.setDefaultMaxPerRoute(25);
+    manager.setMaxTotal(50);
+    SocketConfig socketConfig = SocketConfig.custom()
+                                            .setSoKeepAlive(false)
+                                            .setSoTimeout(60000)
```

Comment:
@djalova 
It doesn't retry even a single time. I tested it. It did retry before the change. 
If you keep socket timeut - it will throw timeout error and that's it
IF you keep socket keep alive - it will endless loop with retry without retrying.

Test it this way:
val d = sc.textFile()
f.count

wait 15 min
val d = sc.textFile()
f.count - it will stuck, timeout error.. or fail to retry

You can check the same with original code and see how it behaved there